app/upgrades/v15/upgrade_test.go

```diff
@@ -77,6 +86,83 @@ func (suite *UpgradeTestSuite) TestMigrateNextPoolIdAndCreatePool() {
 	suite.Require().Equal(gammPoolCreationFee, poolmanagerPoolCreationFee)
 }
 
+
+func (suite *UpgradeTestSuite) TestMigrateBalancerToStablePools() {
+	suite.SetupTest() // reset
+
+	ctx := suite.Ctx
+	gammKeeper := suite.App.GAMMKeeper
+	poolmanagerKeeper := suite.App.PoolManagerKeeper
+	// bankKeeper := suite.App.BankKeeper
+	testAccount := suite.TestAccs[0]
+
+	// Mint some assets to the accounts.
+	suite.FundAcc(testAccount, DefaultAcctFunds)
+
+	// Create the balancer pool
+	swapFee, err := sdk.NewDecFromStr("0.003")
+	exitFee, err := sdk.NewDecFromStr("0.025")
+	poolID, err := suite.App.PoolManagerKeeper.CreatePool(
+		suite.Ctx,
+		balancer.NewMsgCreateBalancerPool(suite.TestAccs[0],
+		balancer.PoolParams{
+			SwapFee: swapFee,
+			ExitFee: exitFee,
+		},
+		[]balancertypes.PoolAsset{
+			{
+				Weight: sdk.NewInt(100),
+				Token:  sdk.NewCoin("foo", sdk.NewInt(5000000)),
+			},
+			{
+				Weight: sdk.NewInt(200),
+				Token:  sdk.NewCoin("bar", sdk.NewInt(5000000)),
+			},
+		},
+		""),
+	)
+	suite.Require().NoError(err)
+
+	// join the pool
+	shareOutAmount := sdk.NewInt(10000000)
+	tokenInMaxs := sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(5000000)), sdk.NewCoin("bar", sdk.NewInt(5000000)))
+	_, _, err = suite.App.GAMMKeeper.JoinPoolNoSwap(suite.Ctx, testAccount, poolID, shareOutAmount, tokenInMaxs)
```

Comment:
So this function has `sharesOut` as one of its return, we could use these to then use for `ExitPool`

Reply:
Nice, used this below